src/daft-parquet/src/read.rs

```diff
@@ -335,7 +348,9 @@ async fn read_parquet_single_into_arrow(
         metadata_num_columns
     };
 
-    if table_ncol != expected_num_columns {
+    if (!field_id_mapping_provided && table_ncol != expected_num_columns)
+        || (field_id_mapping_provided && table_ncol > expected_num_columns)
```

Comment:
shouldn't this be `<=`

Reply:
if `field_id_mapping_provided`, then we expect that we should return strictly fewer number of columns than the expected number of columns.

Thus if the number of columns is `>`, we enter this if statement and throw an error